src/Data/Connection/PostObjectConnectionResolver.php

```diff
@@ -322,20 +327,22 @@ public function get_query_args() {
 					true
 				) ) {
 					$query_args['orderby'] = esc_sql( $orderby_input['field'] );
-				} elseif ( ! empty( $orderby_input['field'] ) ) {
 
-					$order = $orderby_input['order'];
+					// If we're ordering explicitly, there's no reason to check other orderby inputs.
+					break;
+				}
 
-					if ( isset( $query_args['graphql_args']['last'] ) && ! empty( $query_args['graphql_args']['last'] ) ) {
-						if ( 'ASC' === $order ) {
-							$order = 'DESC';
-						} else {
-							$order = 'ASC';
-						}
-					}
+				$order = $orderby_input['order'];
 
-					$query_args['orderby'][ esc_sql( $orderby_input['field'] ) ] = esc_sql( $order );
+				if ( isset( $query_args['graphql_args']['last'] ) && ! empty( $query_args['graphql_args']['last'] ) ) {
+					if ( 'ASC' === $order ) {
+						$order = 'DESC';
+					} else {
+						$order = 'ASC';
+					}
 				}
+
+				$query_args['orderby'][ esc_sql( $orderby_input['field'] ) ] = esc_sql( $order );
```

Comment:
Before the [above change](https://github.com/wp-graphql/wp-graphql/pull/2538/files?diff=split&w=1#r979216342), this was throwing an error since `$query_args['orderby']` could be a `string` if a previous `$orderby_input['field']` was `post__in` etc.